File: frontend/lib/models/expense_record.dart
class ExpenseRecord {
  final String id;
  final double amount;
  final String category;
  final String remark;
  final DateTime recordedAt;
  final DateTime createdAt;

  const ExpenseRecord({
    required this.id,
    required this.amount,
    this.category = '其他',
    this.remark = '',
    required this.recordedAt,
    required this.createdAt,
  });

  factory ExpenseRecord.fromJson(Map<String, dynamic> json) => ExpenseRecord(
    id: json['id'] as String,
    amount: (json['amount'] as num).toDouble(),
    category: json['category'] as String? ?? '其他',
    remark: json['remark'] as String? ?? '',
    recordedAt: DateTime.parse(json['recorded_at'] as String),
    createdAt: DateTime.parse(json['created_at'] as String),
  );

  Map<String, dynamic> toJson() => {
    'amount': amount,
    'category': category,
    'remark': remark,
    'recorded_at': recordedAt.toIso8601String(),
  };
}

File: backend/app/models/expense_record.py
import uuid
from datetime import datetime
from sqlalchemy import String, Numeric, DateTime, ForeignKey, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship
from app.database import Base


class ExpenseRecord(Base):
    __tablename__ = "expense_records"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False, index=True)
    amount: Mapped[float] = mapped_column(Numeric(12, 2), nullable=False)
    category: Mapped[str] = mapped_column(String(20), default="其他")
    remark: Mapped[str] = mapped_column(String(500), default="")
    recorded_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), nullable=False)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

    user = relationship("User", back_populates="expenses")
